brave/src/main/java/com/linecorp/armeria/common/brave/ArmeriaHttpTracingBuilder.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common.brave;
+
+import static java.util.Objects.requireNonNull;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.client.brave.ArmeriaHttpClientParser;
+import com.linecorp.armeria.server.brave.ArmeriaHttpServerParser;
+
+import brave.Tracing;
+import brave.http.HttpClientParser;
+import brave.http.HttpSampler;
+import brave.http.HttpServerParser;
+import brave.http.HttpTracing;
+
+/**
+ * A helper class for creating a new {@link HttpTracing} instance.
+ * For example,
+ *
+ * <pre>{@code
+ * HttpTracing httpTracing = new ArmeriaHttpTracingBuilder(tracing)
+ *     .build();
+ * }</pre>
+ *
+ */
+public class ArmeriaHttpTracingBuilder {
```

Comment:
maybe we need to chat more about this, I don't think it will work if someone needs to provide an armeria specific tracing object, unless you are just trying to match content. For concerns like tags not being available at a specific point in time, we can play a different game.